src/three/TilesRenderer.js

```diff
@@ -865,6 +873,9 @@ export class TilesRenderer extends TilesRendererBase {
 	setTileVisible( tile, visible ) {
 
 		const scene = tile.cached.scene;
+		if(!scene){
+			return;
+		}
```

Comment:
This condition shouldn't be needed

Reply:
It is necessary to keep it because the first parsing of the root subtree doesn't return 3d object so no scene has been initialized.

Reply:
Loading external tile sets also don't produce 3d geometry but this isn't needed for those cases. It's likely related to https://github.com/NASA-AMMOS/3DTilesRendererJS/pull/650#discussion_r1706242631 - if the unrenderable / renderable content flags are set correctly then `setVisible` etc should never be called for geometry that doesn't have 3d geometry.

Reply:
Sorry I've removed my comment...
The comment was a misconception on my side.
(Old comment: ~~It is necessary to keep it because the first parsing of the root subtree doesn't return 3d object so no scene has been initialized or am I missing something here ?~~)

The `preprocessNode` method of my plugin handles only the tileset root, that's why the flags weren't well defined for the children. Thank you :+1:

Reply:
I've fixed it, thank you.
I'm not proud of my condition though. I need something to detect that it's a subtree, without content or with an other subtree as content ...
```js	
	}else if (/.subtree$/i.test(tile.content?.uri) || (tile.__subtreeIdx !== undefined && !uri)){
			tile.__hasUnrenderableContent = true;
			tile.__hasRenderableContent = false;
			
```

Reply:
I'm not fully following how or why the `uri` argument could be null or undefined here. This should be the `tileSetDir` as is passed into the main `preprocessNode` function, no? And why do we need to adjust the content setting based on the `__subTreeIndex` field?

It should be entirely based on the `content` value. If a tile has no content uri then both "has renderable content" and "has unrenderable content" should be false. If it has content that is renderable geometry then only "has renderable content" should be true. And if it has content that points to a file that is used to expand children or somehow load something that _isn't_ renderable (json file, subtree file) then only "has unrenderable content" should be true.